query-engine/core/src/response_ir/internal.rs

```diff
@@ -559,11 +559,6 @@ fn convert_prisma_value(field: &OutputFieldRef, value: PrismaValue, st: &ScalarT
         (ScalarType::Decimal, PrismaValue::Float(f)) => PrismaValue::String(f.to_string()),
 
         (ScalarType::BigInt, PrismaValue::BigInt(i)) => PrismaValue::BigInt(i),
-        (ScalarType::BigInt, PrismaValue::Int(i)) => PrismaValue::BigInt(i),
-        (ScalarType::BigInt, PrismaValue::Float(f)) => PrismaValue::BigInt(
-            f.to_i64()
-                .ok_or_else(|| CoreError::decimal_conversion_error(&f, "i64"))?,
-        ),
 
         (ScalarType::Boolean, PrismaValue::Boolean(b)) => PrismaValue::Boolean(b),
         (ScalarType::Int, PrismaValue::Boolean(b)) => PrismaValue::Int(b as i64),
```

Comment:
We're close to being able to delete this whole function after this PR, right? (I can still see xml coercions for example)

Reply:
Yes, you're right. I just lifted a couple more coercions. We're left with Decimal & XML now. I added some comments as to why they're kept in there for now.